src/test/java/com/ericsson/ei/controller/TestQueryControllerImpl.java

```diff
@@ -0,0 +1,112 @@
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.io.File;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest
+public class TestQueryControllerImpl {
+
+    private static final Logger LOGGER = (Logger) LoggerFactory.getLogger(TestQueryControllerImpl.class);
+    private static final String inputPath = "src/test/resources/AggregatedObject.json";
+    private static final String REQUEST = "testCaseExecutions.testCase.verdict:PASSED,testCaseExecutions.testCase.id:TC5,id:6acc3c87-75e0-4b6d-88f5-b1a5d4e62b43";
+    private static final String QUERY = "{\"criteria\" :{\"testCaseExecutions.testCase.verdict\":\"PASSED\", \"testCaseExecutions.testCase.id\":\"TC5\" }, \"options\" :{ \"id\": \"6acc3c87-75e0-4b6d-88f5-b1a5d4e62b43\"} }";
+    private static String input;
+    private static final String DB_NAME = "MissedNotification";
+    private static final String DB_COLLECTION = "Notification";
+
+    @Value("${server.port}")
+    private String serverPort;
+
+    @Autowired
+    private ProcessQueryParams unitUnderTest;
+
+    @Before
+    public void setUp() {
+        input = FileUtils.readFileAsString(new File(inputPath));
+        try (MongoClient mongoClient = new MongoClient()) {
+            DB db = mongoClient.getDB(DB_NAME);
+            DBCollection collection = db.getCollection(DB_COLLECTION);
+            DBObject dbObjectInput = (DBObject) JSON.parse(input);
+            WriteResult result = collection.insert(dbObjectInput);
+            if (result.wasAcknowledged()) {
+                LOGGER.debug("Data Inserted successfully in both the Collections");
+            }
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage());
+        }
+    }
+
+    @Test
+    public void filterFormParamTest() throws Exception {
+        JSONObject inputJArr = new JSONObject(input);
+        LOGGER.debug("The input string is : " + inputJArr.toString());
+        JSONObject output = null;
+        try {
+            JsonNode inputCriteria = new ObjectMapper().readTree(QUERY);
+            JSONArray result = unitUnderTest.filterFormParam(inputCriteria);
+            output = result.getJSONObject(0);
+            LOGGER.debug("Output for FilterFormParamTest is : " + output.toString());
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+        assertNotNull(output);
+    }
+
+    @Test
+    public void filterQueryParamTest() {
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        request.setServerName("localhost:" + serverPort);
+        request.setRequestURI("/query?request=");
+        request.setQueryString("testCaseExecutions.testCase.verdict:PASSED,testCaseExecutions.testCase.id:TC5,id:6acc3c87-75e0-4b6d-88f5-b1a5d4e62b43");
```

Comment:
what is the purpose of request here? I do not see how it helps you to test the system?

Reply:
In line 109 comparing actual and expected url